manticore/ethereum.py

```diff
@@ -288,6 +288,47 @@ def will_evm_execute_instruction_callback(self, state, instruction, arguments):
                 self.add_finding_here(state, "INVALID instruction")
 
 
+class DetectReentrancy2(Detector):
+    """
+    alert if contract changes the state of storage after a call with >2300 gas to an external address
+    """
+
+    LOCS = 'xx.call_locs'
+
+    def will_open_transaction_callback(self, state, tx):
+        if tx.is_human():
+            state.context[self.LOCS] = []
+
+    def will_evm_execute_instruction_callback(self, state, instruction, arguments):
+        if instruction.semantics == 'CALL':
+            gas = arguments[0]
+            dest_address = arguments[1]
+            msg_sender = state.platform.current_vm.caller
+            pc = state.platform.current_vm.pc
+
+            if not state.can_be_true(gas > 2300):
+                return
+
+            # flag any external call that's going to a symbolic (potentially user controlled)?
+            # potentially bc, it could solve to only 1 address, it could solve to nothing userful
+            # or concretely the sender's address
+            if issymbolic(dest_address):
+                state.context.get(self.LOCS, []).append(pc)
```

Comment:
I think we should also save "gas >2300" with the pc.
`... append((pc, gas>2300)) `